code/game/machinery/_machinery.dm

```diff
@@ -397,12 +434,13 @@ Class Procs:
 /obj/machinery/deconstruct(disassembled = TRUE)
 	if(!(flags_1 & NODECONSTRUCT_1))
 		on_deconstruction()
-		if(component_parts && component_parts.len)
+		if(LAZYLEN(component_parts))
 			spawn_frame(disassembled)
 			for(var/obj/item/I in component_parts)
 				I.forceMove(loc)
-			component_parts.Cut()
-			circuit = null
+				if(circuit == I)
+					circuit = null
```

Comment:
don't need to run this check every time since clearing the circuit var is handled by machinery/Exited()

Reply:
Check removed.